src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php

```diff
@@ -1288,8 +1288,8 @@ public function testNoClassFromGlobalNamespaceClassIdWithLeadingSlash()
 
     public function testNoClassFromNamespaceClassIdWithLeadingSlash()
     {
-        $this->expectException('Symfony\Component\DependencyInjection\Exception\RuntimeException');
-        $this->expectExceptionMessage('The definition for "\Symfony\Component\DependencyInjection\Tests\FooClass" has no class attribute, and appears to reference a class or interface. Please specify the class attribute explicitly or remove the leading backslash by renaming the service to "Symfony\Component\DependencyInjection\Tests\FooClass" to get rid of this error.');
+        $this->expectException('Symfony\Component\DependencyInjection\Exception\InvalidArgumentException');
+        $this->expectExceptionMessage('Service definition "\Symfony\Component\DependencyInjection\Tests\FooClass" has no class, and its name looks like a FQCN but it starts with a backslash; remove the leading backslash.');
```

Comment:
Looks like related to #33092. ping @fabpot

Reply:
From `CheckDefinitionValidityPass`!

Reply:
perhaps im onto something - https://github.com/symfony/symfony/pull/33092#issuecomment-520138148